src/main/java/org/openlmis/integration/dhis2/service/communication/ReferenceDataService.java

```diff
@@ -78,10 +79,40 @@ public PageDto<OrderableDto> findAllOrderables() {
    * @return page of ProcessingPeriodDto objects.
    */
   public PageDto<ProcessingPeriodDto> findAllProcessingPeriods() {
-    return doRequest(PROCESSING_PERIODS_RESOURCE_PATH, ProcessingPeriodDto.class);
+    return doPageRequest(PROCESSING_PERIODS_RESOURCE_PATH, ProcessingPeriodDto.class);
   }
 
-  private <T> PageDto<T> doRequest(String resourcePath, Class<T> clazz) {
+  /**
+   * Get {@link ProcessingPeriodDto} with specific ID from referencedata service.
+   *
+   * @param processingPeriodId ID of processing period to get.
+   * @return the {@link ProcessingPeriodDto} with specific ID.
+   */
+  public ProcessingPeriodDto findProcessingPeriod(UUID processingPeriodId) {
+    try {
+      ResponseEntity<ProcessingPeriodDto> response = restTemplate.exchange(
+              URI.create(serviceUrl + API_URL + PROCESSING_PERIODS_RESOURCE_PATH
+                      + processingPeriodId.toString()),
+              HttpMethod.GET,
+              createEntity(authService.obtainAccessToken(), "Bearer"),
+              new ParameterizedTypeReference<ProcessingPeriodDto>() {}
+      );
+
+      try {
+        return response.getBody();
+      } catch (NullPointerException ex) {
```

Comment:
can you do it differently ? please do not catch NPE

Reply:
I removed the try-catch. In this case, IntelliJ did not display a warning about not handling a possible NullPointerException.